x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -21,89 +21,103 @@ export class JsonIndexFilePicker extends Component {
   state = {
     fileUploadError: '',
     fileParsingProgress: '',
-    fileRef: null
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.fileRef !== this.props.fileRef) {
-      this.setState({ fileRef: this.props.fileRef });
-    }
+  async componentDidMount() {
+    this._isMounted = true;
   }
 
-  _fileHandler = async fileList => {
-    const {
-      resetFileAndIndexSettings, setParsedFile, onFileRemove, onFileUpload,
-      transformDetails, setFileRef, setIndexName
-    } = this.props;
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
 
-    const { fileRef } = this.state;
+  _fileHandler = async fileList => {
+    const fileListArr = Array.from(fileList);
```

Comment:
Why do you convert this to an array? When its used in `_validateAndParse` you could do all the same actions on a list.

Reply:
Originally I changed it for testing, but then things went a different way test-wise. I left it in place for 2 reasons. It's just generally easier to use it as an actual array, and eventually [FileList will likely be an array](https://stackoverflow.com/questions/25333488/why-isnt-the-filelist-object-an-array/39027019)